airflow/contrib/operators/azure_livysparksubmit_operator.py

```diff
@@ -0,0 +1,150 @@
+
+
+class LivySparkSubmitOperator(BaseOperator):
+    """
+    Start a Spark Job on a Azure HDInsight cluster.
+
+    Reference:
+        http://livy.incubator.apache.org/docs/latest/rest-api.html
+
+    :param main_class: main class to be submitted job
+    :type main_class: str
+    :param files:	files to be used in this session
+    :type files:	List[Str]
+
+    :param jars:	jars to be used in this session
+    :type jars:	List[str]
+    :param pyFiles:	Python files to be used in this session
+    :type pyFiles:	List[str]
+    :param driverMemory:	Amount of memory to use for the driver process
+    :type driverMemory:	str
+    :param driverCores:	Number of cores to use for the driver process
+    :type driverCores:	int
+    :param executorMemory:	Amount of memory to use per executor process
+    :type executorMemory:	str
+    :param executorCores:	Number of cores to use for each executor
+    :type executorCores:	int
+    :param numExecutors:	Number of executors to launch for this session
+    :type numExecutors:	int
+    :param archives:	Archives to be used in this session
+    :type archives:	str
+    :param queue:	The name of the YARN queue to which submitted
+    :type queue:	str
+    :param name:	The name of this session
+    :type name:	str
+    :param conf:	Spark configuration properties
+    :type conf:	dict
+    """
+
+    @apply_defaults
+    def __init__(self,
+                 azure_conn_id='azure_hdinsight_default',
+                 application_file=None,
+                 main_class=None,
+                 master=None,
+                 name=None,
+                 jars=[],
+                 py_files=[],
+                 files=[],
+                 executor_memory=None,
+                 executor_cores=None,
+                 num_executors=None,
+                 driver_memory=None,
+                 driver_cores=None,
+                 archives=[],
+                 queue=None,
+                 application_args=[],
+                 conf={},
+                 *args,
+                 **kwargs
+                 ):
+
+        super(LivySparkSubmitOperator, self).__init__(*args, **kwargs)
+
+        self.azure_conn_id = azure_conn_id
+        self.files = files
+        self.pyFiles = py_files
+        self.jars = jars
+        self.name = name
+        self.queue = queue
+        self.archives = archives
+        self.numExecutors = num_executors
+        self.driverCores = driver_cores
+        self.executorCores = executor_cores
+        self.driverMemory = driver_memory
+        self.application_file = application_file
+        self.main_class = main_class
+        self.master = master
+        self.executorMemory = executor_memory
+        self.conf = conf
+        self.application_args = application_args
+
+    def execute(self, context):
+        azure_hook = AzureHDInsightHook(azure_conn_id=self.azure_conn_id)
+        logging.info("Executing LivySparkSubmitOperator ")
```

Comment:
It is necessary? This has already appeared in log. However, if this is not in the logs, this should be added for all operators. Probably in a completely different way.